src/App.tsx

```diff
@@ -1,18 +1,41 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState } from 'react';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+import { Todo } from './react-app-env';
+import { useLocalStorage } from './useLocalStorage';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [todoTitle, setTodoTitle] = useState('');
+
   return (
     <div className="todoapp">
       <header className="header">
         <h1>todos</h1>
 
-        <form>
+        <form
+          onSubmit={(event) => {
+            event.preventDefault();
+            setTodos(
+              [...todos, {
+                id: +new Date(),
+                title: todoTitle,
+                completed: false,
+              }],
+            );
```

Comment:
Create a separate function to handle submit.
Also, get previous todos as a setTodos parameter
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state